implementations/utils/metric.py

```diff
@@ -43,6 +45,12 @@ class TensorBoardScalar:
   step: int
 
 
+class TaskState(enum.Enum):
+  FAILED = "failed"
+  UPSTREAM_FAILED = "upstream_failed"
```

Comment:
What do you think of "`SKIPPED`"? `upstream_failed` makes sense in the context of airflow, but it might not be as clear to consumers.

Reply:
`SKIPPED` sounds better. Will update it.